### PR TITLE
Use Welford's Algorithm for Batch Normalization variance (CPU)

### PR DESCRIPTION
**Pull request by LowikC**
Summary:
Batch Normalization was using a simple two-pass algorithm to compute the variance.
We replace it with Welford's algorithm which is single-pass and has better numeric stability.
This is only the CPU implementation.

### CODE REVIEW

**Comment by SsnL** [comment]
This will conflict with @t-vi ‘a batch norm PR. Since his PR should have been merged a week ago (but is currently waiting for an ONNX test fix) I would suggest waiting until that lands.

File: aten/src/THNN/generic/BatchNormalization.c [hunk 1]
```diff
@@ -27,17 +27,20 @@ void THNN_(BatchNormalization_updateOutput)(
...
 
-      mean = (scalar_t) sum / n;
-      THTensor_(set1d)(save_mean, f, (scalar_t) mean);
-
-      // compute variance per input
-      sum = 0;
       TH_TENSOR_APPLY(scalar_t, in,
-        sum += (*in_data - mean) * (*in_data - mean););
+        scalar_t delta = (*in_data - mean);
+        mean += delta / (cntr + 1);
```
Comment (colesbury) [inline aten/src/THNN/generic/BatchNormalization.c]: This does a division in the hot loop. I suspect this will be quite slow. Do you have any timing numbers for this change?

Comment (LowikC) [reply aten/src/THNN/generic/BatchNormalization.c]: Indeed, I've just checked, and processing time is much higher (+60% for BatchNorm2d on a typical CNN feature map).
I'm closing this PR for now.
